packages/bloc/lib/src/cubit.dart

```diff
@@ -148,12 +147,11 @@ abstract class Cubit<State> extends Stream<State> {
   /// handlers.
   @override
   StreamSubscription<State> listen(
-    void Function(State) onData, {
-    Function onError,
-    void Function() onDone,
-    bool cancelOnError,
+    void Function(State)? onData, {
```

Comment:
It doesn't seem right to allow null here, are you sure you need/want that?

Reply:
I was just trying to conform to the underlying API which appears to allow null (https://github.com/dart-lang/sdk/blob/34e8b4229efec853ae171dc1569c03b1d053c58f/sdk/lib/async/stream.dart#L507)

Reply:
The docs here were/are out of date - we updated [handleError](https://github.com/dart-lang/sdk/blob/34e8b4229efec853ae171dc1569c03b1d053c58f/sdk/lib/async/stream.dart#L696) actually after I presumed you were likely mislead here, but it looks like `listen` also needs to be updated :).

Reply:
Sent out https://dart-review.googlesource.com/c/sdk/+/166460 to update the docs